snuba/query/expressions.py

```diff
@@ -42,6 +44,46 @@ def __iter__(self) -> Iterator[Expression]:
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def accept(self, visitor: ExpressionVisitor[TVisited]) -> TVisited:
```

Comment:
Unfortunately with python and type hints we cannot overload methods thus `accept` has to be implemented in all subclasses.

Reply:
I haven't played around with it much, but does [`typing.overload`](https://docs.python.org/3/library/typing.html#typing.overload) not work here?

Reply:
typing.overload does not do what I mean. typing.overload only allow you to define an alias for a function, in the end you have to collapse all the implementations of the overloaded signatures into one implementation method. Which here would defy the purpose. 
In Java/C++ I would be able to do something like:
```
class Visitor {
    void accept(Column e) {
            ...
    }
    void accept(FunctionCall f) {
           ...
    }
}

And the compiler does the appropriate dispatch, then it becomes useful because we can implement visit at the p[arent class level.